docs/design/autoware-interfaces/components/sensing.md

```diff
@@ -0,0 +1,52 @@
+# Sensing
+
+![Node diagram](images/Sensing-Bus-ODD-Architecture.drawio.svg)
```

Comment:
````suggestion
```mermaid
graph TD
    cmp_drv("Drivers"):::cls_drv
    cmp_loc("Localization"):::cls_loc
    cmp_per("Perception"):::cls_per
    cmp_sen("Sensing"):::cls_sen

    msg_ult("<font size=2><b>Ultrasonic Range Array</b></font size>
    <font size=1>sensor_msgs/Range</font size>"):::cls_drv

    msg_img("<font size=2><b>Camera Image</b></font size>
    <font size=1>sensor_msgs/Image</font size>"):::cls_drv

    msg_ldr("<font size=2><b>Lidar Point Cloud</b></font size>
    <font size=1>sensor_msgs/PointCloud2</font size>"):::cls_drv

    msg_rdr("<font size=2><b>Radar Tracks</b></font size>
    <font size=1>TBD</font size>"):::cls_drv

    msg_gnss("<font size=2><b>GNSS Message</b></font size>
    <font size=1>sensor_msgs/NavSatFix</font size>"):::cls_drv

    msg_ins("<font size=2><b>INS Message</b></font size>
    <font size=1>sensor_msgs/Imu</font size>"):::cls_drv

    msg_gnssvel("<font size=2><b>GNSS Velocity</b></font size>
    <font size=1>geometry_msgs/TwistStamped</font size>"):::cls_drv

    msg_ult_sen("<font size=2><b>Ultrasonic Range Array</b></font size>
    <font size=1>sensor_msgs/Range</font size>"):::cls_sen

    msg_img_sen("<font size=2><b>Camera Image</b></font size>
    <font size=1>sensor_msgs/Image</font size>"):::cls_sen

    msg_pc_combined_rdr("<font size=2><b>Combined Radar Tracks</b></font size>
    <font size=1>TBD</font size>"):::cls_sen

    msg_pc_combined_ldr("<font size=2><b>Combined Lidar Point Cloud</b></font size>
    <font size=1>sensor_msgs/PointCloud2</font size>"):::cls_sen

    msg_ins_sen("<font size=2><b>INS Message</b></font size>
    <font size=1>sensor_msgs/Imu</font size>"):::cls_sen

    msg_gnssvel_sen("<font size=2><b>GNSS Velocity</b></font size>
    <font size=1>geometry_msgs/TwistStamped</font size>"):::cls_sen

    msg_pose_gnss("<font size=2><b>GNSS Pose</b></font size>
    <font size=1>geometry_msgs/PoseWithCovarianceStamped</font size>"):::cls_sen
    

    cmp_drv --> msg_ult --> cmp_sen
    cmp_drv --> msg_img --> cmp_sen
    cmp_drv --> msg_rdr --> cmp_sen
    cmp_drv --> msg_ldr --> cmp_sen
    cmp_drv --> msg_gnss --> cmp_sen
    cmp_drv --> msg_ins --> cmp_sen
    cmp_drv --> msg_gnssvel --> cmp_sen
    
    cmp_sen --> msg_ult_sen
    cmp_sen --> msg_img_sen
    cmp_sen --> msg_ins_sen
    cmp_sen --> msg_gnssvel_sen
    cmp_sen --> msg_pc_combined_rdr
    cmp_sen --> msg_pc_combined_ldr
    cmp_sen --> msg_pose_gnss

    msg_ult_sen --> cmp_per
    msg_img_sen --> cmp_per
    msg_pc_combined_rdr --> cmp_per
    msg_pc_combined_ldr --> cmp_per

    msg_pc_combined_ldr --> cmp_loc
    msg_ins_sen --> cmp_loc
    msg_gnssvel_sen --> cmp_loc
    msg_pose_gnss --> cmp_loc

classDef cls_drv fill:#F8CECC,stroke:#999,stroke-width:1px;
classDef cls_loc fill:#D5E8D4,stroke:#999,stroke-width:1px;
classDef cls_per fill:#FFF2CC,stroke:#999,stroke-width:1px;
classDef cls_sen fill:#FFE6CC,stroke:#999,stroke-width:1px;
```
````

Following from the [original diagram](https://drive.google.com/file/d/1amCZ8sxCuY1isEVhGw99oKeIoqfV25Po/view?usp=sharing), what do you think about using this diagram?

Reply:
You can edit/view it under [here.](https://mermaid.live/edit#pako:eNq9V2Fr2zAQ_StG_bJBSmjZRutlgc1ORqArock2GIagyhdHVJaMpKRNS__7zqmdOJ5dO2NZPhndu3tPJz1JeSJMhUBcEmmaLJypH0gHfyxOZqFevQmIr_kKtAnIW9d1mTDp8A4jFEPMlWJU8EdquZI7IMZ2wAQ0AsegGST7MIzsYAYkwiYgDZfRDoPDgXxBxSaaLYVFVG-upHUMf4RP5_3ebf-7sJoaJTlzbqiMwPmsNV33urf9XncL7b9UKeSe9bG6UXqGlU13k7mH35_4TgSPowoRHo1BU2cU07TMQdRZTgtqEeoK6iseUu2MFcdRT6hleCD_JnOTeN5Kha5UcUNTFVNN2Z1pIWD6xW9FFkljKti-Xk8mzjcw5vB2X9PVhNohf2i32rKKfnT9t-yjeNl63isQdVP_Aegybtvs8ghUDFavX_in99zYiaVxAmErIei5zJ3_23cl86PvaoQcx3sl-oTNmIpvuYSwxgFeFnb-lRVeUSBeV3D0I6G8NtLUrM0RnFLizpxSw39st5TXSBl49cwaI-BgGWnST24XnlpRzalk0KwpLZkry2515_S0n3t6853dvNUg9FszCJ3QDBJtQGnTWoiSpl0p3BF_AvfeG8Vu7NcqBrNTpyYoTX2wsC-rAaUTpRkkakH5rqs4uLdd2L63CtOqDpaENYNEGdQM2z4SC32sDhb6WA3Yzn4_HEgmqDE-zJ3sWnPmXAj3ZHjhDTyvY6xWd-CeXF5eZt-n9zy0C_csefhYSsaCWbL_fnDhvzsoGTuSMw-H5wcyp7POkwcfmpJJh-BVGFMe4sv-KW1RQOwCYgiIi58hzClui4AE8hmhdGnVZC0Zca1eQocsk5Ba8DnF_wQxcedUmO3oIORW6RyZUPlLqRzz_Bt72SwN)

Reply:
[There](https://github.com/beginningfan/autoware-documentation/commit/82c16372fded0ffc982f84a39aa7cffaea2af953) is the first version of the diagram I drew, which is similar to yours. But at the suggestion of [others](https://github.com/autowarefoundation/autoware-documentation/pull/328#discussion_r1139748156), what driver related content was deleted. I am not sure which one is better.

Reply:
From the discussion from this [issue](https://github.com/autowarefoundation/autoware/issues/3466), I think it is better to use mermaid. I have updated Fatih's diagram to match with the rest of the description in this page.

````suggestion
```mermaid
graph TD
    cmp_drv("Drivers"):::cls_drv
    cmp_loc("Localization"):::cls_loc
    cmp_per("Perception"):::cls_per
    cmp_sen("Preprocessors"):::cls_sen

    msg_ult("<font size=2><b>Ultrasonics</b></font size>
    <font size=1>sensor_msgs/Range</font size>"):::cls_drv

    msg_img("<font size=2><b>Camera Image</b></font size>
    <font size=1>sensor_msgs/Image</font size>"):::cls_drv

    msg_ldr("<font size=2><b>Lidar Point Cloud</b></font size>
    <font size=1>sensor_msgs/PointCloud2</font size>"):::cls_drv

    msg_rdr("<font size=2><b>Radar Tracks</b></font size>
    <font size=1>radar_msgs/RadarTracks</font size>"):::cls_drv

    msg_gnss("<font size=2><b>GNSS-INS Position</b></font size>
    <font size=1>sensor_msgs/NavSatFix</font size>"):::cls_drv

    msg_gnssori("<font size=2><b>GNSS-INS Orientation</b></font size>
    <font size=1>autoware_sensing_msgs/GnssInsOrientationStamped</font size>"):::cls_drv

    msg_gnssvel("<font size=2><b>GNSS Velocity</b></font size>
    <font size=1>geometry_msgs/msg/TwistWithCovarianceStamped</font size>"):::cls_drv

    msg_gnssacc("<font size=2><b>GNSS Acceleration</b></font size>
    <font size=1>geometry_msgs/msg/TwistWithCovarianceStamped</font size>"):::cls_drv

    msg_ult_sen("<font size=2><b>Ultrasonics</b></font size>
    <font size=1>sensor_msgs/Range</font size>"):::cls_sen

    msg_img_sen("<font size=2><b>Camera Image</b></font size>
    <font size=1>sensor_msgs/Image</font size>"):::cls_sen

    msg_pc_combined_rdr("<font size=2><b>Combined Radar Tracks</b></font size>
    <font size=1>radar_msgs/RadarTracks</font size>"):::cls_sen

    msg_pc_combined_ldr("<font size=2><b>Combined Lidar Point Cloud</b></font size>
    <font size=1>sensor_msgs/PointCloud2</font size>"):::cls_sen

    msg_pose_gnss("<font size=2><b>GNSS-INS Pose</b></font size>
    <font size=1>geometry_msgs/PoseWithCovarianceStamped</font size>"):::cls_sen

    msg_gnssori_sen("<font size=2><b>GNSS-INS Orientation</b></font size>
    <font size=1>sensor_msgs/Imu</font size>"):::cls_sen

    msg_gnssvel_sen("<font size=2><b>GNSS Velocity</b></font size>
    <font size=1>geometry_msgs/msg/TwistWithCovarianceStamped</font size>"):::cls_sen

    msg_gnssacc_sen("<font size=2><b>GNSS-INS Acceleration</b></font size>
    <font size=1>autoware_sensing_msgs/GnssInsOrientationStamped</font size>"):::cls_sen
    

    cmp_drv --> msg_ult --> cmp_sen
    cmp_drv --> msg_img --> cmp_sen
    cmp_drv --> msg_rdr --> cmp_sen
    cmp_drv --> msg_ldr --> cmp_sen
    cmp_drv --> msg_gnss --> cmp_sen
    cmp_drv --> msg_gnssori --> cmp_sen
    cmp_drv --> msg_gnssvel --> cmp_sen
    cmp_drv --> msg_gnssacc --> cmp_sen
    
    cmp_sen --> msg_ult_sen
    cmp_sen --> msg_img_sen
    cmp_sen --> msg_gnssori_sen
    cmp_sen --> msg_gnssvel_sen
    cmp_sen --> msg_pc_combined_rdr
    cmp_sen --> msg_pc_combined_ldr
    cmp_sen --> msg_pose_gnss
    cmp_sen --> msg_gnssacc_sen

    msg_ult_sen --> cmp_per
    msg_img_sen --> cmp_per
    msg_pc_combined_rdr --> cmp_per
    msg_pc_combined_ldr --> cmp_per

    msg_pc_combined_ldr --> cmp_loc
    msg_pose_gnss --> cmp_loc
    msg_gnssori_sen --> cmp_loc
    msg_gnssvel_sen --> cmp_loc
    msg_gnssacc_sen --> cmp_loc

classDef cls_drv fill:#F8CECC,stroke:#999,stroke-width:1px;
classDef cls_loc fill:#D5E8D4,stroke:#999,stroke-width:1px;
classDef cls_per fill:#FFF2CC,stroke:#999,stroke-width:1px;
classDef cls_sen fill:#FFE6CC,stroke:#999,stroke-width:1px;
```
````

Reply:
There may be a mistake with the type of `GNSS-INS Acceleration` in outputs. I checked the definition of `autoware_sensing_msgs/GnssInsOrientationStamped` in [here](https://github.com/autowarefoundation/autoware_msgs/blob/main/autoware_sensing_msgs/msg/GnssInsOrientationStamped.msg). I don't think it can express acceleration information. @mitsudome-r @xmfcx

Reply:
You are right, it is meant to be AccelerationWithCovarianceStamped.
I have updated my suggestion.

Reply:
Here's the README from sensing_msgs package referring to acceleration: https://github.com/autowarefoundation/autoware_msgs/tree/main/autoware_sensing_msgs#acceleration

Reply:
All done